src/App.js

```diff
@@ -14,11 +15,118 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goodsList: [...goodsFromServer],
+    appStarted: false,
+    isReverse: false,
+    sorting: 'default',
+  }
+
+  setVisible = () => {
+    this.setState(state => ({
+      appStarted: !state.appStarted,
+    }));
+  }
+
+  makeReverse = () => {
+    this.setState(state => ({
+      isReverse: !state.isReverse,
+    }));
+  }
+
+  alphabetSort = () => {
+    this.setState({ sorting: 'alphabet' });
+  }
+
+  lengthSort = () => {
+    this.setState({ sorting: 'length' });
+  }
```

Comment:
this can be one function, which accept sorting method and stores it in state (setSortingMode)